### DISCUSSION TITLE
Questions from testing the mainline environment

### DISCUSSION THREAD

**Discussion by samantha-code** [RaspberryPiFoundation/blockly-keyboard-experimentation#85] [General]
Code.org tested the non-experiment environment (mainline) and had a few questions. I believe some of these questions/ concerns have been mitigated in the experiment version but I wanted to post the questions here for posterity.

About Shift+Control+K:
🐛:Shift+Control+K opens a new tab in Microsoft Edge: If you have the workspace selected, it does put you into keyboard navigation mode on the current tab, but the browser duplicates the tab and puts you into the new duplicated tab. So you have to ctrl-shift-tab to navigate backwards in the tab order to get back. (or ctrl-w to close the duplicated tab)
Is there any concern about the shift-control-k combination being sufficiently accessible enough? It relies upon the user to hold down 3 keys, probably with two hands.

About Default Cursor:
Why did we pick Default as the Default Cursor? Line is better for sighted folks, is default better for non-sighted folks? How should we give the options?

About WASD vs TAB:
Generally seems like issues with conflicting enter targets for Blockly and non-Blockly - I tried to attach a sound block and the Feedback tab suddenly popped open.

When a drop down is down but not selected properly, can a student still use TAB to navigate across the page? Should we freeze until a choice is made? “I need to remember to Blockly-navigate away (WASD) from the block before tabbing over to Run.”
Does WASD work with alternate input devices?

About Block stacks:
Should we have a hotkey to get to the top of the stack?
The default position for a new block is on top of other blocks which is confusing

Thoughts for broader group discussion:
Should we norm on a hotkey for other common access points (for us its run continue workspace)

**Comment by cpcallen** [comment]
Hi @samantha-code, and thanks for your feedback!  A few comments from myself:

> Shift+Control+K

Useful to know that this clashes with an Edge shortcut.

I do not think that we anticipate this being the normal way to activate keyboard navigation; instead I think we anticipate that most users will activate keyboard navigation by using `tab` to navigate to a blockly workspace in the same way that they would access other controls on the page.

> About Default Cursor

As I believe you may now be aware: this is only the default cursor in [the mainline `@blockly/keyboard-nav` plugin](https://github.com/google/blockly-samples/tree/master/plugins/keyboard-navigation).  The new keyboard nav plugin (as is being developed in this repository) uses an improved version of the line cursor.

> About WASD vs TAB

Could you provide a little additional information about what you you were trying to do and how you went about doing so?

There will probably be changes to how navigating the workspace works, including using the arrow keys instead of WASD—if indeed the ability to move the cursor around the open workspace is even needed once we have the ability to move blocks using the keyboard (see #106).

> Should we have a hotkey to get to the top of the stack?

Currently this is possible by pressing the "context out" shortcut (`shift-o`) a few time times, but a dedicated shortcut could be added; feel free to open a feature request in this repo.

> The default position for a new block is on top of other blocks which is confusing.

Agreed, and filed as #95.

> Should we norm on a hotkey for other common access points (for us its run continue workspace)

Could you clarify what you mean by "run continue workspace"?
